CHANGE_LOG

```diff
@@ -1,3 +1,442 @@
+  Namely:
+
+  * ``Generator.beta``
+  * ``Generator.chisquare``
+  * ``Generator.exponential``
+  * ``Generator.f``
+  * ``Generator.gamma``
+  * ``Generator.geometric``
+  * ``Generator.integers``
+  * ``Generator.laplace``
+  * ``Generator.logistic``
+  * ``Generator.lognormal``
+  * ``Generator.logseries``
+  * ``Generator.negative_binomial``
+  * ``Generator.noncentral_chisquare``
+  * ``Generator.noncentral_f``
+  * ``Generator.normal``
+  * ``Generator.pareto``
+  * ``Generator.permutation``
+  * ``Generator.poisson``
+  * ``Generator.power``
+  * ``Generator.random``
+  * ``Generator.rayleigh``
+  * ``Generator.shuffle``
+  * ``Generator.standard_cauchy``
+  * ``Generator.standard_exponential``
+  * ``Generator.standard_gamma``
+  * ``Generator.standard_normal``
+  * ``Generator.standard_t``
+  * ``Generator.triangular``
+  * ``Generator.uniform``
+  * ``Generator.wald``
+  * ``Generator.weibull``
+  * ``Generator.zipf``
+
+* The ``nbytes`` property on NumPy ``ndarray`` types is implemented.
+* Nesting of nested-array types is now supported.
+* ``datetime`` and ``timedelta`` types can be cast to ``int``.
+* ``F``-order iteration is supported in ``ufunc`` generation for increased
+  performance when using combinations of predominantly ``F``-order arrays.
+* The following functions are also now supported:
+
+  * ``np.argpartition``
+  * ``np.isclose``
+  * ``np.new_axis``
+  * ``np.union1d``
+
+Highlights of core changes:
+
+* A large amount of refactoring has taken place to convert many of Numba's
+  internal implementations, of both Python and NumPy functions, from the
+  low-level extension API to the high-level extension API (``numba.extending``).
+* The ``__repr__`` method is supported for Numba types.
+* The default ``target`` for applicable functions in the extension API
+  (``numba.extending``) is now ``"generic"``. This means that ``@overload*`` and
+  ``@intrinsic`` functions will by default be accepted by both the CPU and CUDA
+  targets.
+* The use of ``__getitem__`` on Numba types is now supported in compiled code.
+  i.e. ``types.float64[:, ::1]`` is now compilable.
+
+Performance:
+
+* The performance of ``str.find()`` and ``str.rfind()`` has been improved.
+* Unicode support for ``__getitem__`` now avoids allocation and returns a view.
+* The ``numba.typed.Dict`` dictionary now accepts an ``n_keys`` option to enable
+  allocating the dictionary instance to a predetermined initial size (useful to
+  avoid resizes!).
+* The Numba Run-time (NRT) has been improved in terms of performance and safety:
+
+  * The NRT internal statistics counters are now off by default (removes atomic
+    lock contentions).
+  * Debug cache line filling is off by default.
+  * The NRT is only compiled once a compilation starts opposed to at function
+    decoration time, this improves import speed.
+  * The NRT allocation calls are all made through a "checked" layer by default.
+
+CUDA:
+
+* TBC
```

Comment:
@gmarkall please could you provide the content for any updates made to the CUDA target e.g. via a "suggestion" comment? Many thanks.

Reply:
I've added it in  https://github.com/gmarkall/numba/commit/8c08f9dba5ac3fe08aedaf71018ec204951921d8

Reply:
Many thanks @gmarkall, incorporated as 37ed6e4.